Can't build with clang 10

No problem @woodruffw!
I believe that is more a feature-request than a issue, but happily the project can support both.
I did change the clang version from 9 to 10, sadly I'm out of time during the week but I can take a closer look during the weekend if no one else fix that first. 
```
-- Found LLVM 10.0.1
-- Using LLVMConfig.cmake in: /usr/lib64/cmake/llvm
-- Using ClangConfig.cmake in: /usr/lib/cmake/clang
-- Configuring done
-- Generating done
-- Build files have been written to: /home/patrick/git/constexpr-everything/b
[ 50%] Building CXX object CMakeFiles/constexpr-everything.dir/main.cpp.o
/home/patrick/git/constexpr-everything/main.cpp: In function ‘bool {anonymous}::CheckConstexprDeclStmt(clang::Sema&, const clang::FunctionDecl*, clang::DeclStmt*, clang::SourceLocation&)’:
/home/patrick/git/constexpr-everything/main.cpp:113:49: error: ‘err_constexpr_local_var_no_init’ is not a member of ‘clang::diag’; did you mean ‘ext_constexpr_local_var_no_init’?
  113 |           SemaRef.Diag(VD->getLocation(), diag::err_constexpr_local_var_no_init)
      |                                                 ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
      |                                                 ext_constexpr_local_var_no_init
/home/patrick/git/constexpr-everything/main.cpp: In member function ‘bool ConstexprFunctionASTVisitor::VisitFunctionDecl(clang::FunctionDecl*)’:
/home/patrick/git/constexpr-everything/main.cpp:217:17: error: ‘class clang::Sema’ has no member named ‘CheckConstexprFunctionDecl’; did you mean ‘CheckConstexprFunctionDefinition’?
  217 |       if (!sema.CheckConstexprFunctionDecl(func))
      |                 ^~~~~~~~~~~~~~~~~~~~~~~~~~
      |                 CheckConstexprFunctionDefinition
/home/patrick/git/constexpr-everything/main.cpp:224:17: error: ‘class clang::Sema’ has no member named ‘CheckConstexprFunctionBody’; did you mean ‘CheckConstexprKind’?
  224 |       if (!sema.CheckConstexprFunctionBody(func, func->getBody()))
      |                 ^~~~~~~~~~~~~~~~~~~~~~~~~~
      |                 CheckConstexprKind
/home/patrick/git/constexpr-everything/main.cpp: In member function ‘bool ConstexprVarDeclFunctionASTVisitor::ConstexprVarDeclVisitor::VisitDeclStmt(clang::DeclStmt*)’:
/home/patrick/git/constexpr-everything/main.cpp:283:13: warning: unused variable ‘sema’ [-Wunused-variable]
  283 |       auto &sema = CI_.getSema();
      |             ^~~~
/home/patrick/git/constexpr-everything/main.cpp: At global scope:
/home/patrick/git/constexpr-everything/main.cpp:40:6: warning: ‘bool {anonymous}::CheckConstexprDeclStmt(clang::Sema&, const clang::FunctionDecl*, clang::DeclStmt*, clang::SourceLocation&)’ defined but not used [-Wunused-function]
   40 | bool CheckConstexprDeclStmt(Sema &SemaRef, const FunctionDecl *Dcl,
      |      ^~~~~~~~~~~~~~~~~~~~~~
make[2]: *** [CMakeFiles/constexpr-everything.dir/build.make:82: CMakeFiles/constexpr-everything.dir/main.cpp.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:97: CMakeFiles/constexpr-everything.dir/all] Error 2
make: *** [Makefile:103: all] Error 2
 ✘  ~/git/constexpr-everything/b   m
```

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -38,9 +38,14 @@ set(LIBRARY_LIST
         clangToolingCore
         )
 
-find_package(LLVM 9.0 REQUIRED CONFIG)
+find_package(LLVM REQUIRED CONFIG)
 message(STATUS "Found LLVM ${LLVM_PACKAGE_VERSION}")
 message(STATUS "Using LLVMConfig.cmake in: ${LLVM_DIR}")
+
+if((${LLVM_PACKAGE_VERSION} VERSION_LESS "9.0.0")
+  AND (${LLVM_PACKAGE_VERSION} VERSION_GREATER_EQUAL  "11"))
+  message(FATAL_ERROR "Only LLVM 9 and 10 are supported.")
+endif()
 
 # The clang package doesn't appear to provide a version
 find_package(Clang REQUIRED CONFIG)

--- a/main.cpp
+++ b/main.cpp
@@ -110,8 +110,13 @@ bool CheckConstexprDeclStmt(Sema &SemaRef, const FunctionDecl *Dcl,
           return false;
         if (!VD->getType()->isDependentType() && !VD->hasInit() &&
             !VD->isCXXForRangeDecl()) {
+#if (LLVM_VERSION_MAJOR == 10)
+          SemaRef.Diag(VD->getLocation(), diag::ext_constexpr_local_var_no_init)
+              << isa<CXXConstructorDecl>(Dcl);
+#else
           SemaRef.Diag(VD->getLocation(), diag::err_constexpr_local_var_no_init)
               << isa<CXXConstructorDecl>(Dcl);
+#endif
           return false;
         }
       }
@@ -214,8 +219,14 @@ public:
       std::unique_ptr<int, decltype(returnDiagnostics)> scope(
           &lol, returnDiagnostics);
 
+#if (LLVM_VERSION_MAJOR == 10)
+      if (!sema.CheckConstexprFunctionDefinition(
+              func, Sema::CheckConstexprKind::CheckValid))
+        return true;
+#else
       if (!sema.CheckConstexprFunctionDecl(func))
         return true;
+#endif
 
       // We can't check this if we don't have a function body.
       if (!func->getBody())

